src/library/scala/collection/mutable/ArraySeq.scala

```diff
@@ -184,6 +184,27 @@ object ArraySeq extends StrictOptimizedClassTagSeqFactory[ArraySeq] { self =>
       case that: ofChar => Arrays.equals(array, that.array)
       case _ => super.equals(that)
     }
+
+    override def addString(sb: StringBuilder, start: String, sep: String, end: String): StringBuilder = {
+      val jsb = sb.underlying
+      val len = array.length
+      jsb.append(start)
+      if (len > 0) {
+        if (sep.isEmpty) jsb.append(array)
+        else {
+          jsb.ensureCapacity(jsb.length + len + end.length + (len - 1) * sep.length)
```

Comment:
Not sure what the point of this sizing is given that the elements could be anywhere from 0 to who-knows-how-many characters long, and typically won't be length one, which means there will have to be a reallocation anyway.

Reply:
This is for a char array. Elements are 1 char.

Reply:
Right.  I just looked at the diff, not the full file, so I missed the context.